src/main/java/frc/trigon/robot/subsystems/arm/toohardarm/ToohardArmIO.java

```diff
@@ -0,0 +1,104 @@
+
+        inputs.angleMotorVoltage = masterAngleMotor.getBusVoltage();
+        inputs.anglePositionDegrees = getAnglePositionDegrees();
+        inputs.angleVelocityDegreesPerSecond = getAngleVelocityDegreesPerSecond();
+    }
+
+    @Override
+    protected void setTargetAngleState(TrapezoidProfile.State targetState) {
+        setAngleMotorsPower(calculateAnglePowerFromState(targetState));
+    }
+
+    @Override
+    protected void setTargetElevatorState(TrapezoidProfile.State targetState) {
+        setElevatorMotorsPower(calculateElevatorPowerFromState(targetState));
+    }
+
+    @Override
+    protected void stopAngleMotors() {
+        masterAngleMotor.stopMotor();
+        followerAngleMotor.stopMotor();
+    }
+
+    @Override
+    protected void stopElevatorMotors() {
+        masterElevatorMotor.stopMotor();
+        followerElevatorMotor.stopMotor();
+    }
+
+    private double calculateElevatorPowerFromState(TrapezoidProfile.State targetState) {
+        double pidOutput = ToohardArmConstants.ELEVATOR_PID_CONTROLLER.calculate(
+                getElevatorPositionRevolutions(),
+                targetState.position
+        );
+        double feedforward = ToohardArmConstants.ELEVATOR_FEEDFORWARD.calculate(targetState.velocity);
+        return pidOutput + feedforward;
+    }
+
+    private double calculateAnglePowerFromState(TrapezoidProfile.State targetState) {
+        double pidOutput = ToohardArmConstants.ANGLE_PID_CONTROLLER.calculate(
+                getAnglePositionDegrees(),
+                targetState.position
+        );
+        double feedforward = ToohardArmConstants.ANGLE_FEEDFORWARD.calculate(
+                Units.degreesToRadians(targetState.position),
+                targetState.velocity
+        );
+        return pidOutput + feedforward;
+    }
+
+    private double getElevatorPositionRevolutions() {
+        return Conversions.magTicksToRevolutions(elevatorEncoder.getSelectedSensorPosition());
+    }
+
+    private double getElevatorVelocityRevolutionsPerSecond() {
+        double elevatorVelocityRevolutions = Conversions.magTicksToRevolutions(elevatorEncoder.getSelectedSensorVelocity());
+        return Conversions.perHundredMsToPerSecond(elevatorVelocityRevolutions);
+    }
+
+    private double getAnglePositionDegrees() {
+        return Conversions.revolutionsToDegrees(ToohardArmConstants.ANGLE_ENCODER_POSITION_SIGNAL.refresh().getValue());
+    }
+
+    private double getAngleVelocityDegreesPerSecond() {
+        return Conversions.revolutionsToDegrees(ToohardArmConstants.ANGLE_ENCODER_VELOCITY_SIGNAL.refresh().getValue());
+    }
+
+    private void setAngleMotorsPower(double power) {
+        masterAngleMotor.set(powerToVoltage(power));
+        followerAngleMotor.set(powerToVoltage(power));
+    }
+
+    private void setElevatorMotorsPower(double power) {
+        masterElevatorMotor.set(powerToVoltage(power));
+        followerElevatorMotor.set(powerToVoltage(power));
+    }
```

Comment:
Don't use the power to voltage function, just make the calculations return voltage and use `setVoltage`  (and rename things in this function relative to that change)